### PR TITLE
Fixes2

### LINKED ISSUE #429: Radar minimap is messed up
game dat_wrJJim5sb8
Map is Death Valley

### LINKED ISSUE #430: Saving games fails

### LINKED ISSUE #432: Surveillance image flash doesn't work
Start USA1
game dat_7G5U2pYqAk
These dont "flash" in

### LINKED ISSUE #433: Sentry Drones can't fire
Start USA1
game dat_aq4XkZ70V9

### LINKED ISSUE #434: Stealthed buildings don't render properly.
Start GLA mission 02 (with the GLA stealth general) and spy near their base with the spy ability. The arms dealer and other buildings don't render at all. They are supposed to appear and flash when you do this. Infantry does properly render here.
ss

### PR DESCRIPTION
**Pull request by jonwil**

### CODE REVIEW

**Comment by duncanspumpkin** [comment]
When making your commit message make it start with Fix #number then it will link with the respective issue and auto close on merge.

File: src/platform/w3dengine/client/w3dgameclient.cpp [hunk 1]
```diff
@@ -29,8 +29,8 @@ Display *W3DGameClient::Create_GameDisplay()
     return new W3DDisplay;
 }
 
-void W3DGameClient::Set_Team_Color(int red, int blue, int green)
+void W3DGameClient::Set_Team_Color(int red, int green, int blue)
 {
     W3DStatusCircle::g_needUpdate = true;
-    W3DStatusCircle::g_diffuse = (red << 16) + blue + (green << 8);
+    W3DStatusCircle::g_diffuse = blue + ((green + (red << 8)) << 8);
```
Comment (tomsons26) [inline src/platform/w3dengine/client/w3dgameclient.cpp]: (red << 16) + (green << 8) + blue;

File: src/w3d/renderer/mesh.cpp [hunk 2]
```diff
@@ -343,7 +343,10 @@ void MeshClass::Render_Material_Pass(MaterialPassClass *pass, IndexBufferClass *
 
         MultiListIterator<DX8PolygonRendererClass> it(&(m_model->m_polygonRendererList));
```
Comment (tomsons26) [inline src/w3d/renderer/mesh.cpp:344]: is something missing in this loop? there are Draw_Strip Draw_Triangles calls in wb

File: src/w3d/renderer/mesh.cpp [hunk 3]
```diff
@@ -443,11 +446,14 @@ void MeshClass::Render_Material_Pass(MaterialPassClass *pass, IndexBufferClass *
 
         pass->Install_Materials();
         DX8Wrapper::Set_Index_Buffer(ib, 0);
-        DX8Wrapper::Set_World_Identity();
+        DX8Wrapper::Set_Transform(D3DTS_WORLD, m_transform);
 
         MultiListIterator<DX8PolygonRendererClass> it(&(m_model->m_polygonRendererList));
         while (!it.Is_Done()) {
```
Comment (tomsons26) [inline src/w3d/renderer/mesh.cpp:452]: is something missing in this loop? there are Draw_Strip Draw_Triangles calls in wb

File: src/platform/w3dengine/client/w3dgameclient.cpp (hunk 1 above)
Comment (jonwil) [reply src/platform/w3dengine/client/w3dgameclient.cpp]: fixed.

**Review by tomsons26 — APPROVED** [review]

**Pull request merged**